aten/src/ATen/native/nested/NestedTensorMath.h

```diff
@@ -75,5 +75,7 @@ C10_ALWAYS_INLINE std::pair<int64_t, int64_t> _check_nested_layer_norm_inputs(
   return std::make_pair(M, N);
 }
 
+Tensor reshape_nested(const Tensor& self, IntArrayRef proposed_shape);
```

Comment:
Noob question. Why we do need to add this now? And it seems that reshape_nested is not used anywhere after you changed it to reshape_nested_symint in native_functions.yaml?

Reply:
Sure thing - good question! This was originally added for strided (C++) NTs, and we don't support `nt.sizes()` on those. To expand this to support jagged layout NTs, we need to operate in symbolic shapes land, as we always expect them to have a symbolic shape (i.e. a `SingletonSymInt` for the jagged dimension).

> And it seems that reshape_nested is not used anywhere after you changed it to reshape_nested_symint in native_functions.yaml?

Note that `reshape_nested_symint()` calls the old `reshape_nested()` for the case of strided NT.

Reply:
I see, thanks. Totally missed the usage inside reshape_nested_symint. 
I am wondering why do we need to define "reshape_nested" in the header file while most other functions don't? Does it make any difference?

Reply:
> I am wondering why do we need to define "reshape_nested" in the header file while most other functions don't? Does it make any difference?

The reason for this is because `reshape_nested()` is no longer referenced from `native_functions.yaml`. Functions in the dispatch table there will automatically have a header signature generated.